src/lib/y2network/config_writer/sysconfig.rb

```diff
@@ -35,20 +35,19 @@ def write(config)
         return unless config.routing
 
         write_ip_forwarding(config.routing)
-        # list of devices used in routes
-        devices = config.routing.routes.map { |r| r.interface == :any ? :any : r.interface.name }.uniq
-        devices.each do |dev|
-          routes = config.routing.routes.select { |r| dev == :any || r.interface.name == dev }
 
-          writer = if dev == :any
-            ConfigWriter::SysconfigRoutesWriter.new
+        # list of devices used by routes
+        devices = config.interfaces + [:any]
+
+        devices.each do |dev|
+          routes = find_routes_for(dev, config.routing.routes)
+          file = routes_file_for(dev)
+          if dev != :any && routes.empty?
```

Comment:
please explain in comment why we do not remove file for any

Reply:
hmm, comment explaining what code did, but not why. Why we do not want to remove that `route` file? Is it mandatory to be present on system?

Reply:
I would expect it to be present. At least, we are keeping it currently so I decided to not change the behavior. But it might deserve some documentation.